src/components/fileStorage/repo/files.repo.js

```diff
@@ -0,0 +1,46 @@
+const { FileModel } = require('./db');
+
+/**
+ * @param {*} userId
+ * @return {data} filePermissions
+ */
+const getFilePermissionsByUserId = async (userId) => {
+	return FileModel.aggregate([
+		{
+			$match: {
+				permissions: {
+					$elemMatch: { refId: userId },
+				},
+			},
+		},
+		{
+			$project: {
+				_id: 1,
+				permissions: {
+					$filter: {
+						input: '$permissions',
+						as: 'permission',
+						cond: { $eq: ['$$permission.refId', userId] },
+					},
+				},
+			},
+		},
+	]);
```

Comment:
`lean()` is not needed for aggregate functions, because they already return plain JavaScript objects. Adding `lean()` will actually create an error, because it can't be understood from the JavaScript object.